sdk/src/main/java/io/radar/sdk/RadarHuaweiLocationClient.kt

```diff
@@ -0,0 +1,135 @@
+package io.radar.sdk
+
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.content.Context
+import android.location.Location
+import android.os.Looper
+import com.huawei.hms.location.*
+
+@SuppressLint("MissingPermission")
+internal class RadarHuaweiLocationClient(
+    context: Context,
+    private val logger: RadarLogger
+): RadarAbstractLocationClient() {
+
+    @SuppressLint("VisibleForTests")
+    val locationClient = FusedLocationProviderClient(context)
+    val geofenceService = GeofenceService(context)
+
+    override fun getCurrentLocation(desiredAccuracy: RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy, block: (location: Location?) -> Unit) {
+        val priority = when(desiredAccuracy) {
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.HIGH -> LocationRequest.PRIORITY_HIGH_ACCURACY
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.MEDIUM -> LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.LOW -> LocationRequest.PRIORITY_LOW_POWER
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.NONE -> LocationRequest.PRIORITY_NO_POWER
+        }
```

Comment:
These are also identical in both subclasses, and in `requestLocationUpdates()` below. Can they be factored out to one common function?

Reply:
they're being imported from different packages, though, so while the name is the same, I don't think they can be factored that way.

Reply:
that being said, the priority mapping should be extracted to a method given we're repeating ourselves below, too

Reply:
a lot of the methods here are repeats of the google implementation, though, so if the location clients / classes share the same base classes, then we should push those definitions to the base class

Reply:
That's right. These look identical, but they're from different packages. `RadarHuaweiLocationClient` is using classes from `com.huawei.hms.location.*`, whereas `RadarGoogleLocationClient.kt` is using classes from `com.google.android.gms.location.*`. That's the whole point of factoring them out.